Build failure: linuxPackages_latest.broadcom_sta

## Steps To Reproduce

Steps to reproduce the behavior:

1. build `linuxPackages_latest.broadcom_sta`

## Build log

<!-- insert build log in code block in collapsable section -->

<details>

<summary>Build Log</summary>

```
Running phase: configurePhase
no configure script, doing nothing
Running phase: buildPhase
build flags: SHELL=/nix/store/0irlcqx2n3qm6b1pc9rsd2i8qpvcccaj-bash-5.2p37/bin/bash KBASE=/nix/store/69rmkmic4ifh9xbv6m23cxa8kp7l0328-linux-6.12.1-dev/lib/modules/6.12.1
KBUILD_NOPEDANTIC=1 make -C /nix/store/69rmkmic4ifh9xbv6m23cxa8kp7l0328-linux-6.12.1-dev/lib/modules/6.12.1/build M=`pwd`
make[1]: Entering directory '/nix/store/69rmkmic4ifh9xbv6m23cxa8kp7l0328-linux-6.12.1-dev/lib/modules/6.12.1/build'
CFG80211 API is prefered for this kernel version
Using CFG80211 API
  CC [M]  /build/broadcom-sta/src/shared/linux_osl.o
  CC [M]  /build/broadcom-sta/src/wl/sys/wl_linux.o
/build/broadcom-sta/src/wl/sys/wl_linux.c:59:10: fatal error: asm/unaligned.h: No such file or directory
   59 | #include <asm/unaligned.h>
      |          ^~~~~~~~~~~~~~~~~
compilation terminated.
make[3]: *** [/nix/store/69rmkmic4ifh9xbv6m23cxa8kp7l0328-linux-6.12.1-dev/lib/modules/6.12.1/source/scripts/Makefile.build:229: /build/broadcom-sta/src/wl/sys/wl_linux.o] Error 1
make[2]: *** [/nix/store/69rmkmic4ifh9xbv6m23cxa8kp7l0328-linux-6.12.1-dev/lib/modules/6.12.1/source/Makefile:1936: /build/broadcom-sta] Error 2
make[1]: *** [/nix/store/69rmkmic4ifh9xbv6m23cxa8kp7l0328-linux-6.12.1-dev/lib/modules/6.12.1/source/Makefile:224: __sub-make] Error 2
make[1]: Leaving directory '/nix/store/69rmkmic4ifh9xbv6m23cxa8kp7l0328-linux-6.12.1-dev/lib/modules/6.12.1/build'
make: *** [Makefile:173: all] Error 2
error: builder for '/nix/store/w2wqw6ra75saj7f78n94zg8gk68lwyk1-broadcom-sta-6.30.223.271-6.12.1.drv' failed with exit code 2;
       last 25 log lines:
       > patching file src/wl/sys/wl_iw.c
       > applying patch /nix/store/6xlyf3j0nv4ai7771hf0kbwsvdjl6hmb-source/wl-kmod-026_kernel_6.10_fix_empty_body_in_if_warning.patch
       > patching file src/wl/sys/wl_linux.c
       > applying patch /nix/store/6xlyf3j0nv4ai7771hf0kbwsvdjl6hmb-source/wl-kmod-027_wpa_supplicant-2.11_add_max_scan_ie_len.patch
       > patching file src/wl/sys/wl_cfg80211_hybrid.c
       > Running phase: updateAutotoolsGnuConfigScriptsPhase
       > Running phase: configurePhase
       > no configure script, doing nothing
       > Running phase: buildPhase
       > build flags: SHELL=/nix/store/0irlcqx2n3qm6b1pc9rsd2i8qpvcccaj-bash-5.2p37/bin/bash KBASE=/nix/store/69rmkmic4ifh9xbv6m23cxa8kp7l0328-linux-6.12.1-dev/lib/modules/6.12.1
       > KBUILD_NOPEDANTIC=1 make -C /nix/store/69rmkmic4ifh9xbv6m23cxa8kp7l0328-linux-6.12.1-dev/lib/modules/6.12.1/build M=`pwd`
       > make[1]: Entering directory '/nix/store/69rmkmic4ifh9xbv6m23cxa8kp7l0328-linux-6.12.1-dev/lib/modules/6.12.1/build'
       > CFG80211 API is prefered for this kernel version
       > Using CFG80211 API
       >   CC [M]  /build/broadcom-sta/src/shared/linux_osl.o
       >   CC [M]  /build/broadcom-sta/src/wl/sys/wl_linux.o
       > /build/broadcom-sta/src/wl/sys/wl_linux.c:59:10: fatal error: asm/unaligned.h: No such file or directory
       >    59 | #include <asm/unaligned.h>
       >       |          ^~~~~~~~~~~~~~~~~
       > compilation terminated.
       > make[3]: *** [/nix/store/69rmkmic4ifh9xbv6m23cxa8kp7l0328-linux-6.12.1-dev/lib/modules/6.12.1/source/scripts/Makefile.build:229: /build/broadcom-sta/src/wl/sys/wl_linux.o] Error 1
       > make[2]: *** [/nix/store/69rmkmic4ifh9xbv6m23cxa8kp7l0328-linux-6.12.1-dev/lib/modules/6.12.1/source/Makefile:1936: /build/broadcom-sta] Error 2
       > make[1]: *** [/nix/store/69rmkmic4ifh9xbv6m23cxa8kp7l0328-linux-6.12.1-dev/lib/modules/6.12.1/source/Makefile:224: __sub-make] Error 2
       > make[1]: Leaving directory '/nix/store/69rmkmic4ifh9xbv6m23cxa8kp7l0328-linux-6.12.1-dev/lib/modules/6.12.1/build'
       > make: *** [Makefile:173: all] Error 2
       For full logs, run 'nix log /nix/store/w2wqw6ra75saj7f78n94zg8gk68lwyk1-broadcom-sta-6.30.223.271-6.12.1.drv'.
```

</details>

## Additional context

<!-- Add any other context about the problem here. -->

## Metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.12.0, NixOS, 25.05 (Warbler), 25.05.20241119.23e89b7`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.24.10`
 - nixpkgs: `/run/current-system/nixpkgs`

## Notify maintainers

<!--
Please @ people who are in the `meta.maintainers` list of the offending package or module.
If in doubt, check `git blame` for whoever last touched something.
-->

---

Note for maintainers: Please tag this issue in your PR.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/os-specific/linux/broadcom-sta/default.nix
+++ b/pkgs/os-specific/linux/broadcom-sta/default.nix
@@ -3,6 +3,7 @@
   stdenv,
   fetchurl,
   fetchFromGitHub,
+  fetchpatch2,
   kernel,
 }:
 
@@ -66,7 +67,13 @@ stdenv.mkDerivation {
 
   nativeBuildInputs = kernel.moduleBuildDependencies;
 
-  patches = map (patch: "${rpmFusionPatches}/${patch}") patchset;
+  patches = map (patch: "${rpmFusionPatches}/${patch}") patchset ++ [
+    # Fix for Kernel 6.12 and later (5f60d5f6bbc1)
+    (fetchpatch2 {
+      url = "https://gist.githubusercontent.com/joanbm/20db669eed4d8367a457780747be8cb9/raw/5536ba1354b6b97013530e7345d3bf29e92225b1/broadcom-wl-fix-linux-6.12.patch";
+      hash = "sha256-Y5VgWp0m5tNp8lxDhIg7IodbqyUsJVHHiVIFgP9ioHE=";
+    })
+  ];
 
   makeFlags = [ "KBASE=${kernel.dev}/lib/modules/${kernel.modDirVersion}" ];
 

